src/test/java/com/epam/esm/repository/tag/TagRepositoryImplTest.java

```diff
@@ -1,64 +1,104 @@
 package com.epam.esm.repository.tag;
 
-import com.epam.esm.configuration.DBTestConfig;
 import com.epam.esm.entity.Tag;
-import com.epam.esm.repository.TagRepository;
+import com.epam.esm.repository.CriteriaSpecification;
+import com.epam.esm.repository.NativeSpecification;
+import com.epam.esm.repository.specification.MostWidelyUsedTagByUserOrders;
+import com.epam.esm.repository.specification.TagsByCertificateIdCriteriaSpecifications;
+import com.epam.esm.repository.specification.TagsByNameCriteriaSpecifications;
 import lombok.RequiredArgsConstructor;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
 
-@ExtendWith(SpringExtension.class)
-@ContextConfiguration(classes = {DBTestConfig.class})
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 class TagRepositoryImplTest {
-    private final TagRepository tagRepository;
+  private final TagRepository tagRepository;
+
+  @Test
+  @Transactional
+  void addTag_ShouldReturn_True_Test() {
+    String expectedName = "tag88";
+    Tag tag = new Tag();
+    tag.setName(expectedName);
+
+    String actualTagName = tagRepository.add(tag).get().getName();
+    assertEquals(expectedName, actualTagName);
+  }
+
+  @Test
+  @Transactional
+  void removeTag_ShouldReturn_True_Test() {
+    long expectedId = 9;
+    long actualResult = tagRepository.remove(expectedId).get().getTagId();
+    ;
+    Assertions.assertEquals(expectedId, actualResult);
+  }
+
+  @Test
+  @Transactional
+  void findByName_ShouldReturn_True_Test() {
```

Comment:
Where are negative test cases?

Reply:
Negative case "empty()" is in TagServiceImplTest.java